packages/hub/src/lib/file-download-info.ts

```diff
@@ -4,14 +4,58 @@ import type { CredentialsParams, RepoDesignation } from "../types/public";
 import { checkCredentials } from "../utils/checkCredentials";
 import { toRepoId } from "../utils/toRepoId";
 
+const HUGGINGFACE_HEADER_X_REPO_COMMIT = "X-Repo-Commit"
+const HUGGINGFACE_HEADER_X_LINKED_ETAG = "X-Linked-Etag"
+const HUGGINGFACE_HEADER_X_LINKED_SIZE = "X-Linked-Size"
+
 export interface FileDownloadInfoOutput {
 	size: number;
 	etag: string;
+	commitHash: string | null;
 	/**
 	 * In case of LFS file, link to download directly from cloud provider
 	 */
 	downloadLink: string | null;
 }
+
+/**
+ * Useful when we want to follow a redirection to a renamed repository without following redirection to a CDN.
+ * If a Location header is `/hello` we should follow the relative direct
+ * However we may have full url redirect, on the same origin, we need to properly compare the origin then.
+ * @param params
+ */
+async function followSameOriginRedirect(params: {
+	url: string,
+	method: string,
+	headers: Record<string, string>,
+	/**
+	 * Custom fetch function to use instead of the default one, for example to use a proxy or edit headers.
+	 */
+	fetch?: typeof fetch;
+}): Promise<Response> {
+	const resp = await (params.fetch ?? fetch)(params.url, {
+		method: params.method,
+		headers: params.headers,
+		// prevent automatic redirect
+		redirect: 'manual',
```

Comment:
This doesn't work in the browser

Reply:
I tried the previous version, the etag is never included, as there is a redirect, therefore the headers are empty.

![image](https://github.com/user-attachments/assets/ac052ecf-986d-44c6-80fb-c0bd523f0ae1)

Meaning, that today the function is already not working in the browser ?

Reply:
ah a CDN misconfiguration probably

Reply:
I tried the following function from the devtools console of https://huggingface.co/google-bert/bert-base-uncased

![image](https://github.com/user-attachments/assets/3d2b7e89-02bb-442e-8cbc-7b2256cb7856)

See the fetch function

```js
(await fetch("https://huggingface.co/bert-base-uncased/resolve/dd4bc8b21efa05ec961e3efc4ee5e3832a3679c7/tf_model.h5", {
    method: 'GET',
    headers: {'Range': 'byte=0-0'}
}));
```

I confirm I am getting the right value in Node

![image](https://github.com/user-attachments/assets/462d2aef-b217-4255-a8da-79b6048714f2)

Reply:
yes our CDN CORS headers are probably misconfigured, missing `Access-Control-Expose-Headers`, we'll fix it.

(Did you see my [other comment](https://github.com/huggingface/huggingface.js/pull/1024#pullrequestreview-2434410169)?)